### PR TITLE
Move okhttp-testing-support to Kotlin

### PR DESCRIPTION
**Pull request by yschimke**
Move testing support classes to kotlin also.  Next step to investigating the issue with duplicate classes is probably to move any /java/ modules with just kotlin to /kotlin/

### CODE REVIEW

**Comment by yschimke** [comment]
I started this to investigate a fix for https://github.com/square/okhttp/issues/5902

But I don't think it fixes it.  Would mainly move to have more src/main classes in Kotlin

File: okhttp-testing-support/src/main/java/okhttp3/FakeDns.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  )
+  override fun lookup(hostname: String): List<InetAddress> {
+    requestedHosts.add(hostname)
+    val result = hostAddresses[hostname]
+    if (result != null) return result
+    throw UnknownHostException()
+  }
+
+  fun assertRequests(vararg expectedHosts: String?) {
+    Assertions.assertThat(requestedHosts)
```
Comment (swankjesse) [inline okhttp-testing-support/src/main/java/okhttp3/FakeDns.kt]: nit: static import `Assertions.*`

File: okhttp-testing-support/src/main/java/okhttp3/RecordingCookieJar.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  fun assertResponseCookies(vararg cookies: String?) {
+    val actualCookies = takeResponseCookies()
+    val actualCookieStrings: MutableList<String> =
+      ArrayList()
+    for (cookie in actualCookies) {
+      actualCookieStrings.add(cookie.toString())
+    }
+    Assertions.assertThat(actualCookieStrings)
```
Comment (swankjesse) [inline okhttp-testing-support/src/main/java/okhttp3/RecordingCookieJar.kt]: same

File: okhttp-testing-support/src/main/java/okhttp3/TestUtil.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  /**
+   * See FinalizationTester for discussion on how to best trigger GC in tests.
+   * https://android.googlesource.com/platform/libcore/+/master/support/src/test/java/libcore/
+   * java/lang/ref/FinalizationTester.java
+   */
+  @Throws(Exception::class)
+  @JvmStatic
+  fun awaitGarbageCollection() {
+    Runtime.getRuntime()
+        .gc()
```
Comment (swankjesse) [inline okhttp-testing-support/src/main/java/okhttp3/TestUtil.kt]: oooh this wrapping hurts me

File: okhttp-testing-support/src/main/java/okhttp3/UppercaseRequestInterceptor.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  override fun intercept(chain: Chain): Response {
+    return chain.proceed(uppercaseRequest(chain.request()))
+  }
+
+  /** Returns a request that transforms `request` to be all uppercase.  */
+  private fun uppercaseRequest(request: Request): Request {
+    val uppercaseBody: RequestBody = object : ForwardingRequestBody(request.body) {
+      @Throws(
+          IOException::class
+      ) override fun writeTo(sink: BufferedSink) {
```
Comment (swankjesse) [inline okhttp-testing-support/src/main/java/okhttp3/UppercaseRequestInterceptor.kt]: sadly there’s no prize for most unique formatting of a `@Throws` clause

**Comment by yschimke** [comment]
Sorry should have put as draft and committed to those fixes before landing. Lazy of me

**Comment by yschimke** [comment]
My PR is so bad it broke github 
image

**Comment by yschimke** [comment]
> Although I made many style nits please don’t fix the formatting; let’s get a robot to do it for us. And we can be confident that we hired the right robot by how it fixed this code.

This is Intellijs code cleanup and format robots

https://github.com/yschimke/okhttp/pull/4/files

**Comment by swankjesse** [comment]
The IntelliJ bug where java.* imports are last is so awful. I think Spotless does a better job.

**Comment by yschimke** [comment]
@swankjesse Do you want to literally land this in ugly form? Then have.a formatter bakeoff?  Or just go with the Intellij (branch) then Spotless (our build checks) duo

**Comment by swankjesse** [comment]
I’d like it to look nice before merge, just because it’s easier to keep a tidy code base than to get one. But if this is blocking you let’s merge it right away.

Have you tried spotlessApply?

**Comment by yschimke** [comment]
I don't think it fixes the problem. Intellij and gradle can build differently. Only fails Android testing.

File: okhttp/src/main/java/okhttp3/HttpUrl.kt [hunk 5]
```diff
@@ -1178,8 +1178,8 @@ class HttpUrl internal constructor(
           password = encodedPassword.percentDecode(),
           host = host ?: throw IllegalStateException("host == null"),
           port = effectivePort(),
-          pathSegments = encodedPathSegments.percentDecode() as List<String>,
-          queryNamesAndValues = encodedQueryNamesAndValues?.percentDecode(plusIsSpace = true),
+          pathSegments = encodedPathSegments.map { it.percentDecode() },
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/HttpUrl.kt:1181]: accidental cleanup

**Pull request merged**